Scripts/Spells: Implemented druid's Skull Bash

matanshukry opened the pull request:
**Changes proposed:**

- Implemented druid's skull bash spell

**Target branch(es):** 3.3.5/master

- [ ] 3.3.5
- [x] master

**Issues addressed:**

Closes #19727

**Tests performed:**

(Does it build, tested in-game, etc.)

- [x] It builds
- [x] Tested in-game

**Known issues and TODO list:** (add/remove lines as needed)

matanshukry committed:
Implemented druid's skull bash spell

offl commented:
#19727?

matanshukry committed:
Merge branch 'master' into issue/skull-bash

illfated commented:
Would it be interesting to sort the enum DruidSpells list, or should we ignore it? I notice it got skipped in your Prayer of Mending PR.

matanshukry commented:
> Would it be interesting to sort the enum DruidSpells list, or should we ignore it? I notice it got skipped in your Prayer of Mending PR.

I'm planning another PR for paladin spells, was gonna sneak the priest enum into that.
I need to fix this one enum anyway, so might as well do it together as well.

Shauren commented:
This is incorrect (wrong sequencing and wrong spell ids)
Using 106839 immediately triggers two spells: 221514 (charge) and 93985 (interrupt). Interrupt does not happen when charge finishes movement (this is a general player interrupt thing, no interrupt of any class has travel time)

matanshukry committed:
- Ordered spells enum alphabetically
- Changed spell ids
- Moved interrupt cast to start of the cast rather than after the charge

matanshukry committed:
Merge remote-tracking branch 'origin/master' into issue/skull-bash

matanshukry commented:
Fixed

Shauren commented on src/server/scripts/Spells/spell_druid.cpp:
```diff
@@ -1541,6 +1543,28 @@ class spell_dru_wild_growth : public SpellScriptLoader
+
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_DRUID_SKULL_BASH_CHARGE });
```
Missing SPELL_DRUID_SKULL_BASH_INTERRUPT here

Shauren commented on src/server/scripts/Spells/spell_druid.cpp:
```diff
@@ -1573,4 +1597,5 @@ void AddSC_druid_spell_scripts()
     new spell_dru_t10_restoration_4p_bonus();
     new spell_dru_t10_restoration_4p_bonus_dummy();
     new spell_dru_wild_growth();
+    RegisterSpellScript(spell_dru_skull_bash);
```
sort by class name

Shauren commented on src/server/scripts/Spells/spell_druid.cpp:
```diff
@@ -1541,6 +1543,28 @@ class spell_dru_wild_growth : public SpellScriptLoader
 };
 
+// 106839 - Skull Bash
+class spell_dru_skull_bash : public SpellScript
```
scripts are sorted (i dont remember why anymore, but they are)

matanshukry committed:
Sorted druid spells file, including register calls and classes orders

Shauren committed:
rename sql
